ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/sftp_outbound_transfer.py

```diff
@@ -0,0 +1,484 @@
+        logger.error("Invalid event notification, no records found: ", exc_info=True)
+        return
+
+    try:
+        event_type_str = s3_event["eventName"]
+        event_type: S3EventType
+        if S3EventType.OBJECT_CREATED in event_type_str:
+            event_type = S3EventType.OBJECT_CREATED
+        else:
+            logger.error("Event type %s is unsupported. Exiting...", event_type_str)
+            return
+    except KeyError:
+        logger.error(
+            "The incoming event record did not contain the type of S3 event: ", exc_info=True
+        )
+        return
+
+    try:
+        file_key: str = s3_event["s3"]["object"]["key"]
+        decoded_file_key = unquote(file_key)
+    except KeyError:
+        logger.error("No bucket file found in event notification: ", exc_info=True)
+        return
+
+    # Log the various bits of data extracted from the invoking event to aid debugging:
+    logger.info("Invoked at: %s UTC", datetime.utcnow().isoformat())
+    logger.info("S3 Object Key: %s", decoded_file_key)
+    logger.info("S3 Event Type: %s, Specific Event Name: %s", event_type.name, event_type_str)
+
+    object_key_pattern = rf"^{BUCKET_ROOT_DIR}/([\w_]+)/([\w_]+)/(.*)$"
+    match = re.search(
+        pattern=object_key_pattern,
+        string=decoded_file_key,
+        flags=re.IGNORECASE,
+    )
+    if match is None:
+        logger.error(
+            "Invocation event unsupported, object key does not match expected pattern: %s. See log"
+            " for additional detail. Exiting...",
+            object_key_pattern,
+        )
+        return
+
+    partner = match.group(1)
+    subfolder = match.group(2)
+    filename = match.group(3)
+
+    try: 
+        global_config = GlobalSsmConfig.from_ssm()
+    except (ValueError, botocore.exceptions.ClientError):
+        logger.error(
+            "An unrecoverable error occurred when attempting to retrieve "
+            "global outbound SSM configuration: ",
+            exc_info=True,
+        )
+        return
+
+    try:
+        partner_config = PartnerSsmConfig.from_partner(partner=partner)
+    except (ValueError, botocore.exceptions.ClientError):
+        logger.error(
+            "An unrecoverable error occurred when attempting to retrieve SSM configuration for"
+            " partner %s: ",
+            partner,
+            exc_info=True,
+        )
+        return
+
+    if subfolder != partner_config.pending_files_dir:
+        logger.error(
+            "%s pending files directory, %s, does not match object key: %s, exiting...",
+            partner,
+            partner_config.pending_files_dir,
+            decoded_file_key,
+        )
+        return
+
+    destination_folder = next(
+        (
+            recognized_file.destination_folder
+            for recognized_file in partner_config.recognized_files
+            if re.search(pattern=recognized_file.filename_pattern, string=filename)
+        ),
+        None,
+    )
```

Comment:
For anyone unaware, `next(...)` is being used like other language's `first(...)` or `take(1)`. Essentially, we just want the first element that matches the filter.